e2e/cat_test.go

```diff
@@ -305,4 +316,150 @@ func TestCatByVersionID(t *testing.T) {
+}
+
+func TestCatPrefix(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	testCases := []struct {
+		files    []string
+		prefix   string
+		expected string
+	}{
+		{files: []string{"file1.txt", "file2.txt"}, prefix: "", expected: "content0content1"},
+		{files: []string{"dir/file3.txt", "dir/file4.txt"}, prefix: "", expected: "content0content1"},
+		{files: nil, prefix: "dir/", expected: "content2content3"},
+		{files: []string{"dir/nesteddir/file5.txt"}, prefix: "dir/", expected: "content2content3"},
+		{files: nil, prefix: "dir/nesteddir/", expected: "content4"},
+	}
+
+	offset := 0
+	for _, tc := range testCases {
+		if tc.files != nil {
+			var concatenatedContent strings.Builder
+			for idx, file := range tc.files {
+				content := fmt.Sprintf("content%d", idx+offset)
+				putFile(t, s3client, bucket, file, content)
+				concatenatedContent.WriteString(content)
+			}
+			offset += len(tc.files)
+		}
+		verifyCatCommand(t, s5cmd, bucket, tc.expected, tc.prefix)
+	}
+}
+
+func TestCatWildcard(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	files := []struct {
+		key     string
+		content string
+	}{
+		{"foo1.txt", "content0"},
+		{"foo2.txt", "content1"},
+		{"bar1.txt", "content2"},
+		{"foolder/foo3.txt", "content3"},
+		{"log-file-2024-01.txt", "content4"},
+		{"log-file-2024-02.txt", "content5"},
+		{"log-file-2023-01.txt", "content6"},
+		{"log-file-2022-01.txt", "content7"},
+	}
+
+	for _, file := range files {
+		putFile(t, s3client, bucket, file.key, file.content)
+	}
+
+	testCases := []struct {
+		prefix   string
+		expected string
+	}{
+		{"foo*", "content0content1content3"},
+		{"log-file-2024-*", "content4content5"},
+		{"log-file-*", "content7content6content4content5"},
+	}
+
+	for _, tc := range testCases {
+		verifyCatCommand(t, s5cmd, bucket, tc.expected, tc.prefix)
+	}
+}
+
+func TestPrefixWildcardFail(t *testing.T) {
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	testCases := []struct {
+		prefix string
+	}{
+		{"foo*"},
+		{"foolder/"},
+	}
+
+	for _, tc := range testCases {
+		cmd := s5cmd("cat", fmt.Sprintf("s3://%v/%v", bucket, tc.prefix))
+		result := icmd.RunCmd(cmd)
+
+		result.Assert(t, icmd.Expected{ExitCode: 1})
+		assertLines(t, result.Stderr(), map[int]compareFunc{
+			0: equals(`ERROR "cat s3://%v/%v": no object found`, bucket, tc.prefix),
+		}, strictLineCheck(false))
+	}
+
+	for _, tc := range testCases {
+		cmd := s5cmd("--json", "cat", fmt.Sprintf("s3://%v/%v", bucket, tc.prefix))
+		result := icmd.RunCmd(cmd)
+
+		result.Assert(t, icmd.Expected{ExitCode: 1})
+		assertLines(t, result.Stderr(), map[int]compareFunc{
+			0: equals(`{"operation":"cat","command":"cat s3://%v/%v","error":"no object found"}`, bucket, tc.prefix),
+		}, strictLineCheck(false))
+	}
```

Comment:
you can combine both tests in single loop and define two subtests, something like ;
```go
for _, tc := testcases {
    t.Run(tc.name, func(t * testing.T){
        t.Run("default", func(t *testing.T){
            cmd := /* ... */
            // ...
        })


        t.Run("json", /* ... */)
    })
}
```

Reply:
Done.